lib/components/card/LoginCard.tsx

```diff
@@ -0,0 +1,172 @@
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+    defaultValues: {
+      userName: "",
+      userPassword: "",
+    },
+  });
+
+  const buttonActionWrapper = (values: z.infer<typeof formSchema>) => {
+    buttonAction(values);
+  };
+
+  const [showPassword, setShowPassword] = useState(false);
+
+  const togglePasswordVisibility = () => {
+    setShowPassword((prev) => !prev);
+  };
+
+  return (
+    <ShadcnCard {...props}>
+      <CardContent
+        className={cn(
+          "sm:max-w-md",
+          "bg-neutral-100 dark:bg-neutral-800",
+          "text-black dark:text-white",
+        )}
+      >
+        <CardHeader>
+          <CardTitle>Welcome to SDA</CardTitle>
+        </CardHeader>
+
+        <ShadcnForm {...form}>
+          <form onSubmit={form.handleSubmit(buttonActionWrapper)}>
+            <div className={cn("mt-medium mb-medium")}>
+              <FormField
+                control={form.control}
+                name="userName"
+                render={({ field }) => (
+                  <FormItem>
+                    <FormLabel>Username *</FormLabel>
+
+                    <FormControl>
+                      <ShadcnInput
+                        className={cn(
+                          "text-neutral-900",
+                          form.formState.errors.userName
+                            ? "border-error-500"
+                            : "border-neutral-300",
+                        )}
+                        placeholder="Enter your username"
+                        {...field}
+                      />
+                    </FormControl>
+
+                    <FormMessage className={cn("text-error-500")} />
+                  </FormItem>
+                )}
+              />
+            </div>
+
+            <div className={cn("mt-medium mb-medium")}>
+              <FormField
+                control={form.control}
+                name="userPassword"
+                render={({ field }) => (
+                  <FormItem>
+                    <FormLabel>Password *</FormLabel>
+                    <FormControl>
+                      <div className={cn("relative")}>
+                        <ShadcnInput
+                          className={cn(
+                            "text-neutral-900",
+                            form.formState.errors.userPassword
+                              ? "border-error-500"
+                              : "border-neutral-300",
+                          )}
+                          type={showPassword ? "text" : "password"}
+                          placeholder="Enter your password"
+                          {...field}
+                        />
+                        <button
+                          type="button"
+                          onClick={togglePasswordVisibility}
+                          className={cn(
+                            "absolute right-2 top-1/2 transform -translate-y-1/2",
```

Comment:
For the show password button, I think the best option is to use the 'absolute' position so that the button stays within the input field container and doesn't displace other form elements. @alebg